### PR TITLE
Bump cairo-vm

### PR DESCRIPTION
**Pull request by notlesh**
Bump `cairo-vm` to latest `main`.

This also required bumping `cairo` to `2.8.2` which required bumping our `rustc` version.

This pulls in an [upstream fix](https://github.com/lambdaclass/cairo-vm/pull/1820) which makes [this hack](https://github.com/Moonsong-Labs/cairo-vm/pull/46) obsolete.

Issue Number: N/A

## Type

- [ ] feature
- [x] bugfix
- [ ] dev (no functional changes, no API changes)
- [ ] fmt (formatting, renaming)
- [ ] build
- [ ] docs
- [ ] testing

## Breaking changes?

- [ ] yes
- [x] no

### CODE REVIEW

**Review by ftheirs — APPROVED** [review]

File: crates/starknet-os/src/sharp/pie.rs [hunk 1]
```diff
@@ -71,7 +71,7 @@ pub fn decode_base64_to_unzipped(pie_str: &str, dst: &str) -> Result<(), SnOsErr
         general_purpose::STANDARD.decode(pie_str.as_bytes()).map_err(|e| SnOsError::PieZipping(format!("{e}")))?;
```
Comment (HermanObst) [inline crates/starknet-os/src/sharp/pie.rs:71]: What are these changes? @notlesh

File: Cargo.toml [hunk 2]
```diff
@@ -30,12 +30,12 @@ base64 = "0.21.3"
...
-cairo-lang-starknet-classes = { version = "=2.7.1" }
-cairo-lang-utils = { version = "=2.7.1" }
-cairo-lang-casm = { version = "=2.7.1" }
+cairo-lang-starknet = { version = "=2.8.2" }
+cairo-lang-starknet-classes = { version = "=2.8.2" }
+cairo-lang-utils = { version = "=2.8.2" }
+cairo-lang-casm = { version = "=2.8.2" }
 cairo-type-derive = { version = "0.1.0", path = "crates/cairo-type-derive" }
-cairo-vm = { git = "https://github.com/Moonsong-Labs/cairo-vm", rev = "56b68b50944ecb3123a168218ea7b8b8e23f9be8", features = ["cairo-1-hints", "extensive_hints", "mod_builtin"] }
+cairo-vm = { git = "https://github.com/Moonsong-Labs/cairo-vm", branch = "notlesh/segment-arena-relocation-fix", features = ["cairo-1-hints", "extensive_hints", "mod_builtin"] }
```
Comment (HermanObst) [inline Cargo.toml:38]: Did you bump this branch to cairo-vm latest? 
@notlesh

**Review by HermanObst — APPROVED** [review]
LGTM
Just some questions to clarify

**Comment by shamsasari** [comment]
I'll wait for this PR before raising mine.

File: Cargo.toml (hunk 2 above)
Comment (notlesh) [reply Cargo.toml:38]: Yes, it the change here basically does:

1. undo the previous hack mentioned in the PR description (by not including the commit 56b68b5)
2. include `merge main` which pulls in the upstream fix (also mentioned in the PR desc)

**Pull request merged**